lib/on_open.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+var on_open = function(haxfred) {
+  var channels = _getSlackChannels(haxfred.slack.channels);
+  var groups = _getSlackGroups(haxfred.slack.groups);
+
+  _initialGreeting(haxfred, channels, groups);
+
+  haxfred.emit('slack.open',{
+    channels: channels,
+    groups: groups
+  });
+};
+
+function _getSlackChannels(channels) {
```

Comment:
(total nitpick)

Since these are not exported by the module, I don't think there's a need to `_` here, since that acts as a means of "privitizing" them.